src/crystal/system/unix/event_loop_libevent.cr

```diff
@@ -174,14 +169,23 @@ class Crystal::LibEvent::EventLoop < Crystal::EventLoop
           raise ::Socket::Error.from_errno("accept")
         end
       else
-        {% unless LibC.has_method?(:accept4) %}
-          Crystal::System::Socket.fcntl(client_fd, LibC::F_SETFD, LibC::FD_CLOEXEC)
-        {% end %}
         return client_fd
       end
     end
   end
 
+  private def accept_cloexec(socket)
+    {% if LibC.has_method?(:accept4) %}
+      LibC.accept4(socket.fd, nil, nil, LibC::SOCK_CLOEXEC)
+    {% else %}
+      Crystal::System::Process.disable_fork do
+        fd = LibC.accept(socket.fd, nil, nil)
+        Crystal::System::Socket.fcntl(fd, LibC::F_SETFD, LibC::FD_CLOEXEC) unless fd == -1
+        fd
+      end
```

Comment:
**issue**: the socket may be in blocking mode, in which case we musn't prevent fork as it may potentially take forever until there is an incoming connection on the socket.

We could check `socket.blocking?` and only block when false _but_ another thread could change it back into blocking mode before we call `#accept` :sob: